### ISSUE TITLE
When iPhone is connected via USB cable, DNS configuration is not applied correctly

### ISSUE THREAD

**Issue by ospfranco** [Tunnelblick/Tunnelblick#746]
**Describe the bug**
I'm using a VPN connection via Tunnelblick. Normally, I can connect without issues, all the configuration is applied correctly and I can browse normally. That is until I plug in my iPhone.

When the iPhone is plugged in at the end of the configuration step I get the following output:

```
2023-03-13 08:43:45.345299 *Tunnelblick: Warning: DNS server address 172.16.100.1 is not being used.


2023-03-13 08:43:45.381738 *Tunnelblick: Warning: DNS server address 172.16.100.2 is not being used.


2023-03-13 08:43:45.382642 *Tunnelblick: Warning: DNS server address 192.168.0.1 is being used but should not be used. That may indicate that more than one network interface is active. Tunnelblick does not support multiple active network interfaces.


2023-03-13 08:43:50.875133 *Tunnelblick: This computer's apparent public IP address (188.195.158.156) was unchanged after the connection was made
```

Things I've tried:
- Checking "Disable secondary network interfaces"
- Deleted the iPhone USB connection.
- Any manually set DNS configuration is removed
- Change a bunch of configurations on iPhone/macOS, disable hand-off, made sure private relay is off, made sure hotspot is turned off

<img width="715" alt="CleanShot 2023-03-13 at 08 51 19" src="https://user-images.githubusercontent.com/1634213/224639217-0e03671f-2fd2-4254-8ecc-29947aecf7da.png">

If the connection has already been established and then I plug in my iPhone then the VPN routes still work, but I cannot access the rest of the internet.

**Desktop (please complete the following information):**
 - OS: macOS 13.1
 - Browser Orion/Safari/Brave
 - Version Tunnelblick 3.8.8a (build 5776)

**Additional context**
I do not know if I should strip any important information from the connection logs, let me know and I can paste it here.

**Comment by ospfranco** [comment]
I just stumbled into this discussion in a google group, it seems indeed the iPhone connection causes some change in the interfaces that are not reflected in the system settings

https://groups.google.com/g/tunnelblick-discuss/c/v0_GBKT1dag

**Comment by ospfranco** [comment]
I tried the following things mentioned in the thread from my last comment:

```
Thank you for your suggestions. I was more radical and followed this suggestion: 

Navigate to the /Library/Preferences/SystemConfiguration/ folder and delete: 

com.apple.airport.preferences.plist 
preferences.plist 

rebooted and now with the default network location can connect without any DNS issue. 
```

After reboot nothing changed, DNS still cannot be set if iPhone is connected. Also reading around I found out about the `scutil -dns` command which produces this output:

```
scutil --dns
DNS configuration

resolver #1
  search domain[0] : hitronhub.home
  nameserver[0] : 192.168.0.1
  if_index : 17 (en0)
  flags    : Request A records
  reach    : 0x00020002 (Reachable,Directly Reachable Address)

resolver #2
  domain   : local
  options  : mdns
  timeout  : 5
  flags    : Request A records
  reach    : 0x00000000 (Not Reachable)
  order    : 300000

resolver #3
  domain   : 254.169.in-addr.arpa
  options  : mdns
  timeout  : 5
  flags    : Request A records
  reach    : 0x00000000 (Not Reachable)
  order    : 300200

resolver #4
  domain   : 8.e.f.ip6.arpa
  options  : mdns
  timeout  : 5
  flags    : Request A records
  reach    : 0x00000000 (Not Reachable)
  order    : 300400

resolver #5
  domain   : 9.e.f.ip6.arpa
  options  : mdns
  timeout  : 5
  flags    : Request A records
  reach    : 0x00000000 (Not Reachable)
  order    : 300600

resolver #6
  domain   : a.e.f.ip6.arpa
  options  : mdns
  timeout  : 5
  flags    : Request A records
  reach    : 0x00000000 (Not Reachable)
  order    : 300800

resolver #7
  domain   : b.e.f.ip6.arpa
  options  : mdns
  timeout  : 5
  flags    : Request A records
  reach    : 0x00000000 (Not Reachable)
  order    : 301000

DNS configuration (for scoped queries)

resolver #1
  search domain[0] : hitronhub.home
  nameserver[0] : 192.168.0.1
  if_index : 17 (en0)
  flags    : Scoped, Request A records
  reach    : 0x00020002 (Reachable,Directly Reachable Address)
```

So indeed the DNS configuration is not set after VPN connects.

**Comment by ospfranco** [comment]
Finally got it working. I went into the 3 dot menu in the macOS network settings and selected `manage virtual interfaces`, there a `Thunderbolt bridge` was available. I deleted it, not sure if it was the best idea but at that point I had nothing else too lose. But now when Tunnelblick starts the connection I no longer get DNS errors. Everything seems to be working just fine.

As for what does the `Thunderbolt bridge` interface do, I cannot really tell... maybe it was some remmant from some UTM virtual machine I had installed some time ago.
